src/index.js

```diff
@@ -50,7 +51,7 @@ function syncReduxAndRouter(history, store, selectRouterState = SELECT_STATE) {
     // Avoid dispatching an action if the store is already up-to-date,
     // even if `history` wouldn't do anything if the location is the same
     if(getRouterState().path !== locationToString(location)) {
-      store.dispatch(updatePath(locationToString(location)));
+      store.dispatch(updatePath(locationToString(location), true));
```

Comment:
When `history.listen` is called the transition to the new route already happened, so a new `pushState` is redundant. We only need an update to the state. Route changes triggered externally through actions/reducer will reset the flag (unless set explicitly in the action)